tree/tree/src/TTreeCacheUnzip.cxx

```diff
@@ -958,7 +872,7 @@ Int_t TTreeCacheUnzip::UnzipBuffer(char **dest, char *src)
       /* early consistency check */
       UChar_t *bufcur = (UChar_t *) (src + keylen);
       Int_t nin, nbuf;
-      if(R__unzip_header(&nin, bufcur, &nbuf)!=0) {
+      if(objlen > nbytes-keylen && R__unzip_header(&nin, bufcur, &nbuf)!=0) {
```

Comment:
This line is changed in a commit titled "Use TTaskGroup interface to unzip baskets in parallel. "
The link between this change and the title is not clear.  If it is not related can you put it in its own commit?